### PR TITLE
Modernize the plugin to java 11

### PR DESCRIPTION
**Pull request by AayushSaini101**
Modernize the plugin to java 11.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,17 +14,8 @@
...
-    <!-- Java Level to use. Java 7 required when using core >= 1.612 -->
-    <java.level>7</java.level>
-    <!-- Jenkins Test Harness version you use to test the plugin. -->
-    <!-- For Jenkins version >= 1.580.1 use JTH 2.x or higher. -->
-    <jenkins-test-harness.version>2.1</jenkins-test-harness.version>
-    <!-- Other properties you may want to use:
-         ~ hpi-plugin.version: The HPI Maven Plugin version used by the plugin..
-         ~ stapler-plugin.version: The Stapler Maven plugin version required by the plugin.
-    -->
+     <java.level>8</java.level>
```
Comment (MarkEWaite) [inline pom.xml:17]: The most recent parent pom no longer requires `java.level`.  I've updated the document to show that correction.
```suggestion
```

**Comment by MarkEWaite** [comment]
Thanks very much!

When I use your change, there is a test that fails if I run `mvn clean verify`.  I believe that the test failure can be resolved by using the plugin bill of materials as described in the Contributing to Open Source document and in the pre-release of the tutorial.  Would you like to make that change in the pull request to use the plugin bill of materials?

Would you also be willing to add a Jenkinsfile in the pull request so that a job will be available on ci.jenkins.io to monitor this repository?

**Comment by MarkEWaite** [comment]
@AayushSaini101 I'd be very grateful if you'd be willing to experiment with the tutorial that is being assembled from the "Contributing to Open Source" document.  The tutorial pull request is available at https://github.com/jenkins-infra/jenkins.io/pull/4806 .

When the pull request build completes, a preview of the tutorial will be available at https://deploy-preview-4806--jenkins-io-site-pr.netlify.app/doc/developer/ in the section labeled "Improve a Plugin - Tutorial".  We expect that the tutorial will have completed review and will be live on https://www.jenkins.io/doc/developer/ before the end of September 2022.

**Comment by AayushSaini101** [comment]
> Thanks very much!
> 
> When I use your change, there is a test that fails if I run `mvn clean verify`. I believe that the test failure can be resolved by using the plugin bill of materials as described in the Contributing to Open Source document and in the pre-release of the tutorial. Would you like to make that change in the pull request to use the plugin bill of materials?
> 
> Would you also be willing to add a Jenkinsfile in the pull request so that a job will be available on ci.jenkins.io to monitor this repository?

new
@MarkEWaite  Thank you very much Sir for reviewing my PR.  When I run `mvn clean verify`. All the test cases passed successfully.  Including the  <java.level>8</java.level> as well excluding the  <java.level>8</java.level>.

**Comment by MarkEWaite** [comment]
Thanks for sharing the example.  I see the same success on Windows as you see.  When I run the tests on Linux, I see the test failure.  I can't explain why that test would be platform specific.  That is quite surprising.

The removal of `java.level` is optional.  I suspect it won't be made mandatory for at least a few more releases of the parent pom.  No objection from me if you leave it in, though there will be a future release of the parent pom that likely mandates the removal of the `java.level` property.

**Comment by AayushSaini101** [comment]
Sure, sir, You can push the bill of materials.

**Pull request merged**